framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/deployment/LdapDeploymentImpl.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.cloud.openshift.deployment;
+
+import org.kie.cloud.api.deployment.LdapDeployment;
+import org.kie.cloud.openshift.resource.Project;
+
+public class LdapDeploymentImpl extends OpenShiftDeployment implements LdapDeployment {
+
+    private static final String HOST_TEMPLATE = "ldap://%s:30389";
+    private static final String DEPLOYMENT_CONFIG_NAME = "ldap";
+
+    private String serviceName;
+    private String host;
+
+    public LdapDeploymentImpl(Project project) {
+        super(project);
+    }
+
+    @Override
+    public String getServiceName() {
+        if (serviceName == null) {
+            serviceName = ServiceUtil.getLdapServiceName(getOpenShift());
+        }
+        return serviceName;
+    }
+
+    @Override
+    public String getDeploymentConfigName() {
+        return DEPLOYMENT_CONFIG_NAME;
+    }
+
+    @Override
+    public String getHost() {
+        if (host == null) {
+            host = String.format(HOST_TEMPLATE, getServiceName());
```

Comment:
It may be better to retrieve ldap port using OpenShift too, to avoid hardcoding values and their duplicity.

Reply:
I didn't want to overcomplicate this logic, but I changed it as you suggested.